### ISSUE TITLE
Cannot multiply I by a variable and add it to a matrix.

### ISSUE THREAD

**Issue by araujoms** [jump-dev/MutableArithmetics.jl#206]
If I create a variable `@variable(model, x)` and a numerical matrix `a = randn(Float64,(2,2))`, the expression `a+x*I` gives me the error `MethodError: no method matching +(::Matrix{Float64}, ::AffExpr)`.

Of course, it works without any problem if instead of `I` one uses an explicit identity matrix like `id = Symmetric(Matrix{Float64}(I,2,2))`

**Comment by odow** [comment]
JuMP now gives a better error here too: https://github.com/jump-dev/JuMP.jl/pull/3558

**Comment by odow** [comment]
Closing because I think this is resolved by #205 and jump-dev/JuMP.jl#3558.

```julia
julia> using JuMP, LinearAlgebra

julia> model = Model();

julia> @variable(model, x);

julia> a = rand(2, 2);

julia> @expression(model, a + x * I)
ERROR: Addition between an array and a JuMP scalar is not supported: instead of `x + y`, do `x .+ y` for element-wise addition. If you are modifying the diagonal entries of a square matrix, do `x + y * LinearAlgebra.I(n)`, where `n` is the side length.
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:35
 [2] _throw_operator_error(#unused#::typeof(+), x::Matrix{Float64})
   @ JuMP ~/.julia/dev/JuMP/src/operators.jl:467
 [3] +(#unused#::AffExpr, x::Matrix{Float64})
   @ JuMP ~/.julia/dev/JuMP/src/operators.jl:485
 [4] muladd(x::VariableRef, y::UniformScaling{Bool}, z::Matrix{Float64})
   @ Base.Math ./math.jl:1430
 [5] add_mul(a::Matrix{Float64}, b::VariableRef, c::UniformScaling{Bool})
   @ MutableArithmetics ~/.julia/packages/MutableArithmetics/solfs/src/MutableArithmetics.jl:27
 [6] operate(op::typeof(MutableArithmetics.add_mul), x::Matrix{Float64}, y::VariableRef, args::UniformScaling{Bool})
   @ MutableArithmetics ~/.julia/packages/MutableArithmetics/solfs/src/interface.jl:210
 [7] macro expansion
   @ ~/.julia/packages/MutableArithmetics/solfs/src/rewrite.jl:321 [inlined]
 [8] macro expansion
   @ ~/.julia/dev/JuMP/src/macros.jl:721 [inlined]
 [9] top-level scope
   @ REPL[7]:1

julia> @expression(model, a + x * I(2))
2×2 Matrix{AffExpr}:
 x + 0.664436809229147  0.05946109185455417
 0.5136992768834855     x + 0.8550754973050526

julia> a + x * I
ERROR: Addition between an array and a JuMP scalar is not supported: instead of `x + y`, do `x .+ y` for element-wise addition. If you are modifying the diagonal entries of a square matrix, do `x + y * LinearAlgebra.I(n)`, where `n` is the side length.
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:35
 [2] _throw_operator_error(#unused#::typeof(+), x::Matrix{Float64})
   @ JuMP ~/.julia/dev/JuMP/src/operators.jl:467
 [3] +(x::Matrix{Float64}, #unused#::AffExpr)
   @ JuMP ~/.julia/dev/JuMP/src/operators.jl:486
 [4] top-level scope
   @ REPL[9]:1

julia> a + x * I(2)
2×2 Matrix{AffExpr}:
 x + 0.664436809229147  0.05946109185455417
 0.5136992768834855     x + 0.8550754973050526
```

**Comment by araujoms** [comment]
Honestly I'm more surprised that LinearAlgebra can do such a thing. At least now it's clear that JuMP is interpreting `I` as a 1x1 identity instead of a 2x2.
